File: PadelClub.Services/Database/Membership.cs
using System;

namespace PadelClub.Services.Database
{
    public class Membership
    {
        public int Id { get; set; }
        public int UserId { get; set; }
        public string MembershipType { get; set; } = string.Empty; // Basic, Premium, VIP
        public DateTime StartDate { get; set; }
        public DateTime EndDate { get; set; }
        public decimal Price { get; set; }
        public bool IsActive { get; set; } = true;
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime? UpdatedAt { get; set; }
        public string Status { get; set; } = "Active";
        public bool AutoRenew { get; set; }
        public bool CancelAtPeriodEnd { get; set; }
        public DateTime? SuspendedAt { get; set; }
        public DateTime? CancelledAt { get; set; }

        // Navigation properties
        public virtual User User { get; set; } = null!;
        public virtual Payment? Payment { get; set; }
        public virtual ICollection<MembershipEvent> Events { get; set; } = new List<MembershipEvent>();
    }
}

File: UI/padelclub_desktop/lib/features/memberships/domain/entities/membership.dart
class Membership {
  final int? id;
  final int userId;
  final String membershipType;
  final DateTime startDate;
  final DateTime endDate;
  final double price;
  final bool isActive;
  final DateTime createdAt;
  final DateTime? updatedAt;

  Membership({
    this.id,
    required this.userId,
    required this.membershipType,
    required this.startDate,
    required this.endDate,
    required this.price,
    required this.isActive,
    required this.createdAt,
    this.updatedAt,
  });
}
